flixel/input/gamepad/ButtonIndex.hx

```diff
@@ -0,0 +1,405 @@
+			default: -1;
+		}
+	}
+	
+	public inline function getRawXbox(buttonID:ButtonID):Int
+	{
+		return switch(buttonID)
+		{
+			case A: XboxButtonID.A;
+			case B: XboxButtonID.B;
+			case X: XboxButtonID.X;
+			case Y: XboxButtonID.Y;
+			case BACK: XboxButtonID.BACK;
+			case GUIDE: XboxButtonID.XBOX;
+			case START: XboxButtonID.START;
+			case LEFT_STICK_BTN: XboxButtonID.LEFT_STICK_BTN;
+			case RIGHT_STICK_BTN: XboxButtonID.RIGHT_STICK_BTN;
+			case LEFT_SHOULDER: XboxButtonID.LB;
+			case RIGHT_SHOULDER: XboxButtonID.RB;
+			case DPAD_UP: XboxButtonID.DPAD_UP;
+			case DPAD_DOWN: XboxButtonID.DPAD_DOWN;
+			case DPAD_LEFT: XboxButtonID.DPAD_LEFT;
+			case DPAD_RIGHT: XboxButtonID.DPAD_RIGHT;
+			
+			case LEFT_TRIGGER:  XboxButtonID.LEFT_TRIGGER;
+			case RIGHT_TRIGGER: XboxButtonID.RIGHT_TRIGGER;
+			
+			default: -1;
+		}
+	}
+	
+	public inline function getRawXInput(buttonID:ButtonID):Int
+	{
+		return switch(buttonID)
+		{
+			case A: XInputButtonID.A;
+			case B: XInputButtonID.B;
+			case X: XInputButtonID.X;
+			case Y: XInputButtonID.Y;
+			case BACK: XInputButtonID.BACK;
+			case GUIDE: XInputButtonID.GUIDE;
+			case START: XInputButtonID.START;
+			case LEFT_STICK_BTN: XInputButtonID.LEFT_STICK_BTN;
+			case RIGHT_STICK_BTN: XInputButtonID.RIGHT_STICK_BTN;
+			case LEFT_SHOULDER: XInputButtonID.LB;
+			case RIGHT_SHOULDER: XInputButtonID.RB;
+			case DPAD_UP: XInputButtonID.DPAD_UP;
+			case DPAD_DOWN: XInputButtonID.DPAD_DOWN;
+			case DPAD_LEFT: XInputButtonID.DPAD_LEFT;
+			case DPAD_RIGHT: XInputButtonID.DPAD_RIGHT;
+			
+			case LEFT_TRIGGER: XInputButtonID.LEFT_TRIGGER;
+			case RIGHT_TRIGGER: XInputButtonID.RIGHT_TRIGGER;
+			
+			//case LEFT_TRIGGER: 
+			//case RIGHT_TRIGGER: 
+			default: -1;
+		}
+	}
+	
+	/**************/
+	
+	public inline function getBtnOUYA(rawID:Int):ButtonID
+	{
+		return switch(rawID)
+		{
+			case OUYAButtonID.O: A;
+			case OUYAButtonID.A: B;
+			case OUYAButtonID.U: X;
+			case OUYAButtonID.Y: Y;
+			case OUYAButtonID.HOME: GUIDE;
+			case OUYAButtonID.LEFT_STICK_BTN: LEFT_STICK_BTN;
+			case OUYAButtonID.RIGHT_STICK_BTN: RIGHT_STICK_BTN;
+			case OUYAButtonID.LB: LEFT_SHOULDER;
+			case OUYAButtonID.RB: RIGHT_SHOULDER;
+			//case BACK:
+			//case START:
+			//case DPAD_UP:
+			//case DPAD_DOWN:
+			//case DPAD_LEFT:
+			//case DPAD_RIGHT:
+			default: NONE;
+		}
+	}
+	
+	public inline function getBtnLogitech(rawID:Int):ButtonID
+	{
+		return switch(rawID)
+		{
+			case LogitechButtonID.TWO: A;
+			case LogitechButtonID.THREE: B;
+			case LogitechButtonID.ONE: X;
+			case LogitechButtonID.FOUR: Y;
+			case LogitechButtonID.NINE: BACK;
+			case LogitechButtonID.TEN: START;
+			case LogitechButtonID.LEFT_STICK_BTN: LEFT_STICK_BTN;
+			case LogitechButtonID.RIGHT_STICK_BTN: RIGHT_STICK_BTN;
+			case LogitechButtonID.FIVE: LEFT_SHOULDER;
+			case LogitechButtonID.SIX: RIGHT_SHOULDER;
+			case LogitechButtonID.SEVEN: LEFT_TRIGGER;
+			case LogitechButtonID.EIGHT: RIGHT_TRIGGER;
+			//case DPAD_UP: 
```

Comment:
Cool, I'll fill out the logic and if you could run the gamepad demo with your logitech controller (perhaps putting some traces in FlxInput to trace out what hardware ID's are being pressed) and deduce what each button input should be, that'd be swell.